cms/djangoapps/contentstore/management/commands/tests/test_cleanup_assets.py

```diff
@@ -23,7 +24,8 @@ class ExportAllCourses(ModuleStoreTestCase):
     def setUp(self):
         """ Common setup. """
         self.content_store = contentstore()
-        self.module_store = modulestore()
+        # pylint: disable=protected-access
+        self.module_store = modulestore()._get_modulestore_by_type(ModuleStoreEnum.Type.mongo)
```

Comment:
There are couple of different issues.

1) The "_id" field for each asset differs in type between split and old mongo. For old mongo, it is the dict form of the location (hence the asset['_id']['name'] check), but for split it is the unicode representation of the locator. This could be worked around in the test by checking the name of the asset in a different way, though perhaps code should change to make "_id" consistent.

2) course.id.make_asset_key("asset", None) throws a TypeError for split. I don't know what to change the 2nd argument to in order to keep the desired behavior the same. Given the difference in type for "_id", I am also concerned that the code further down won't work--
query['_id.name'] = all_assets[0]['_id']['name']
and
asset_doc['_id']['name'] = u'._example_test.txt'
self.content_store.fs_files.insert(asset_doc)